src/main/java/cinema/util/HashUtil.java

```diff
@@ -0,0 +1,27 @@
+package cinema.util;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
+    public static String hashPassword(String password, byte[] salt) {
+        MessageDigest messageDigest;
+        try {
+            messageDigest = MessageDigest.getInstance("SHA-512");
+            messageDigest.update(salt);
+            byte[] hash = messageDigest.digest(password.getBytes());
+            return new String(hash, StandardCharsets.UTF_8);
```

Comment:
let's use StringBuilder and for each byte - append the following:

            for (byte b: digest) {
                hashedPassword.append(String.format("%02x", b));
            }

Reply:
fixed